docs/design/fabric-v2+/attestation-interface.md

```diff
@@ -0,0 +1,63 @@
+# Attestation Interface
+
+This document defines the interface for producing and verifying an attestation.
+
+
+## Producing an attestation
+
+This interface is only available inside an enclave.
+
+```
+int init_attestation(
+    uint8_t* params,
+    uint32_t params_length);
+
+```
+```
+int get_attestation(
+    uint8_t statement[32],
+    uint8_t* attestation,
+    uint32_t attestation_length);
+```
+
+
+### Details related to EPID-based SGX attestations
+
+FPC can use EPID-based SGX attestations.
+Such attestations have some peculiarities.
+1. They require to be initialized with some external parameters (SPID, signature revocation list). These parameters are provided through `init_attestation`.
+
+2. They are computed indirectly through a different enclave, called Quoting Enclave. For this reason, the implementation of `get_attestation` uses the following edge functions to retrieve the IAS-verifiable attestation.
+```
+void ocall_init_quote(
+    [out, size=target_len] uint8_t *target, uint32_t target_len,
+    [out, size=egid_len] uint8_t *egid, uint32_t egid_len);
+
+void ocall_get_quote(
+    [in, size=spid_len] uint8_t *spid, uint32_t spid_len,
+    [in, size=report_len] uint8_t *report, uint32_t report_len,
+    [out, size=max_quote_len] uint8_t *quote, uint32_t max_quote_len,
+    [out] uint32_t *actual_quote_len);
+```
+
+3. It must to be sent to IAS, which provides a publicly verifiable report. This step thus involves contacting and authenticating with a web service using the ISV's credentials. For this reason, its completion is delegated to a different entity.
```

Comment:
But still would involve a, yet to be defined, ocall? 
BTW: thinking of how to get the credentials, i actually realized that the current way of passing it via decorators should work in this case (with all its warts, of course, but at least one viable way to manage the credentials)

Reply:
If the attestation output has to be publicly verifiable, then yes: we need another ocall and the IAS credentials passed somehow.

If the attestation output can be IAS-verifiable (i.e, the spid owner has to convert this into a publicly verifiable report by contacting IAS), then no: we don't need ocall and credentials; we delegate this to the admin.

The second case is viable since the admin is already the recipient of the `createenclave` response.